base/inference.jl

```diff
@@ -1181,23 +1234,37 @@ function abstract_call(f::ANY, fargs, argtypes::Vector{Any}, vtypes::VarTable, s
             end
         end
         return Any
+    elseif f === UnionAll
+        if length(fargs) == 3 && isa(argtypes[2],Const)
+            if isType(argtypes[3])
+                body = argtypes[3].parameters[1]
+            elseif isa(argtypes[3],Const)
+                body = argtypes[3].val
+            else
+                return Any
```

Comment:
I think there can be cases where it returns a typevar or possibly something else; being cautious.